STYLEGUIDE.md

```diff
@@ -0,0 +1,326 @@
+# Ember.js JavaScript Style Guide
+
+## Table Of Contents
+
++ [Objects](#objects)
++ [Array](#arrays)
++ [Variables](#variables)
++ [Whitespace](#whitespace)
++ [Commas](#commas)
++ [Semicolons](#semicolons)
++ [Block Statements](#block-statements)
++ [Conditional Statements](#conditional-statements)
++ [Properties](#properties)
++ [Functions](#functions)
++ [Function Arguments](#function-arguments)
++ [Comments](#comments)
+
+## Objects
+
++ use literal form for object creation.
+
+```javascript
+var foo = {};
+```
+
+## Arrays
+
++ use literal form for array creation (unless you know the exact length).
+
+```javascript
+var foo = [];
+```
+
++ if you know the exact length and know that array is not going to grow, use `Array`.
+
+```javascript
+var foo = new Array(16);
+```
+
++ use `push` to add an item to an array.
+
+```javascript
+var foo = [];
+foo.push('bar');
+```
+
+## Variables
+
++ put all declarations on one line.
+
+```javascript
+var a, b;
+```
+
++ use multiple lines to declare and assign variables.
+
+```javascript
+var a = 1;
+var b = 2;
+```
+
++ declare variables at the top of their scope.
+
+```javascript
+function foo() {
+  var bar;
+
+  console.log('foo bar!');
+
+  bar = getBar();
+}
+```
+
+## Whitespace
+
++ use soft tabs set to 2 spaces.
+
+```javascript
+function() {
+∙∙var name;
+}
+```
+
++ place 1 space before the leading brace.
+
+```javascript
+obj.set('foo', {
+  foo: 'bar'
+});
+
+test('foo-bar', function() {
+});
+```
+
++ no spaces before semicolons.
+
+```javascript
+var foo = {};
+```
+
++ no spaces after the function name when it's declared or called.
+
+```javascript
+function foo() {
+}
+
+foo();
+```
+
+## Commas
+
++ no trailing commas.
+
+```javascript
+var foo = [1, 2, 3];
+var bar = {a: 'a'};
+```
+
++ no leading commas.
+
+```javascript
+var foo = [
+  1
+  , 2
```

Comment:
This example doesn't follow the rest (all of the rest so far are showing the "right" way to do things but this shows the wrong way).